src/dispatchers/abstractDispatcher.ts

```diff
@@ -3,21 +3,22 @@
 module Plottable {
   export class Dispatcher extends Core.PlottableObject {
     protected _event2Callback: { [eventName: string]: (e: Event) => any; } = {};
-    protected _broadcasters: Core.Broadcaster<Dispatcher>[] = [];
+    protected _callbackSets: Utils.CallbackSet<Function>[] = [];
     private _connected = false;
 
     private _hasNoListeners() {
-      return this._broadcasters.every((b) => b.getListenerKeys().length === 0);
+      return this._callbackSets.every((cbs) => cbs.values().length === 0);
     }
 
     private _connect() {
-      if (!this._connected) {
-        Object.keys(this._event2Callback).forEach((event: string) => {
-          var callback = this._event2Callback[event];
-          document.addEventListener(event, callback);
-        });
-        this._connected = true;
+      if (this._connected) {
+        return;
       }
+      Object.keys(this._event2Callback).forEach((event: string) => {
```

Comment:
Never mind, apparently `Object.keys` is vastly superior:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys